src/runtime/native/runtime.ml

```diff
@@ -101,24 +113,35 @@ let dump_counters_exn =
   Common.write_runtime_data
 
 let dump () =
-  match file_channel () with
-  (* Failure "TypeError: runtime.unix_open is not a function" indicates that the
-     code was compiled by js_of_ocaml. In this case, we *want* to do nothing at
-     process exit. See https://github.com/aantron/bisect_ppx/issues/334. *)
-  | exception Failure _ -> ()
-  | None -> ()
-  | Some channel ->
-      (try
-        dump_counters_exn channel
-      with _ ->
-        verbose Unable_to_write_file);
-      close_out_noerr channel
+  if not !bisect_file_written then
+    (if !sigterm_enable then
+       ignore @@ Unix.(sigprocmask SIG_BLOCK [Sys.sigterm]) ;
+     bisect_file_written := true;
+     match file_channel () with
+     (* Failure "TypeError: runtime.unix_open is not a function" indicates that the
+        code was compiled by js_of_ocaml. In this case, we *want* to do nothing at
+        process exit. See https://github.com/aantron/bisect_ppx/issues/334. *)
+     | exception Failure _ -> ()
+     | None -> ()
+     | Some channel ->
+       (try
+          dump_counters_exn channel
+        with _ ->
+          verbose Unable_to_write_file);
+       close_out_noerr channel)
+
+let sigterm_handler (_ : int) =
+  dump ();
+  exit 0
 
 let register_dump : unit Lazy.t =
   lazy (at_exit dump)
 
-let register_file ~bisect_file ~bisect_silent ~filename ~points =
+let register_file ~bisect_file ~bisect_silent ~bisect_sigterm ~filename ~points =
   (match bisect_file with None -> () | Some v -> default_bisect_file := v);
   (match bisect_silent with None -> () | Some v -> default_bisect_silent := v);
+  sigterm_enable := env_to_boolean ~default:bisect_sigterm "BISECT_SIGTERM";
+  (if !sigterm_enable then
```

Comment:
I believe this will currently install the signal handler each time a source file is registered with the Bisect runtime. It should probably be guarded by some `ref` so that the signal handler is installed only once.

Reply:
In 3d413f8 I wrap the registration in a lazy, like is done for the `at_exit` registration. Is that OK?

Reply:
Yes, thank you!